liblangutil/Scanner.h

```diff
@@ -90,12 +90,16 @@ class Scanner
 {
 	friend class LiteralScope;
 public:
-	explicit Scanner(CharStream _source = CharStream(), std::string _sourceName = "") { reset(std::move(_source), std::move(_sourceName)); }
+	explicit Scanner(std::shared_ptr<CharStream> _source) { reset(std::move(_source)); }
+	explicit Scanner(CharStream _source = CharStream()) { reset(std::move(_source)); }
 
-	std::string source() const { return m_source.source(); }
+	std::string source() const { return m_source->source(); }
```

Comment:
Should perhaps return by `const&`?

Reply:
I noticed that too. But I didn't want to change too much that's not explicitly related to SourceLocation->CharStream refactor. Since seperate PR for smallies like that don't make much sense, I'll do that.